src/app/modules/OpenshiftStreams/dialogs/CreateInstance/CreateInstanceWithSizes/hooks/useAvailableProvidersAndDefault.ts

```diff
@@ -0,0 +1,136 @@
+import { useAuth, useConfig } from '@rhoas/app-services-ui-shared';
+import { Configuration, DefaultApi } from '@rhoas/kafka-management-sdk';
+import { InstanceType } from '@app/utils';
+import { convertQuotaToInstanceType, getQuotaType, useAMSQuota } from './useAMSQuota';
+
+/**
+ * Hooks for fetching available providers and their regions
+ *
+ * @returns
+ */
+export const useAvailableProvidersAndDefault = () => {
+  const { kas, getUsername } = useAuth();
+  const {
+    kas: { apiBasePath: basePath },
+  } = useConfig();
+  const getQuota = useAMSQuota();
+
+  function getApi() {
+    return new DefaultApi(
+      new Configuration({
+        accessToken: kas.getToken(),
+        basePath,
+      })
+    );
+  }
+
+  // Function to fetch cloud Regions based on selected filter
+  const fetchRegions = async (
+    id: string,
+    instance_type: string
+  ): Promise<CreateKafkaInstanceWithSizesTypes.Regions> => {
+    const apisService = getApi();
+    const res = await apisService.getCloudProviderRegions(id);
+
+    if (!res?.data?.items) {
+      return [];
+    }
+
+    const regionsForInstance = res.data.items.filter(
+      (p) => p.enabled && p.capacity.some((c) => c.instance_type === instance_type)
+    );
+
+    return regionsForInstance.map((r): CreateKafkaInstanceWithSizesTypes.RegionInfo => {
+      const max_capacity_reached = r.capacity?.some((c) => c.available_sizes?.length === 0);
+
+      return {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        id: r.id!,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        displayName: r.display_name!,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        isDisabled: max_capacity_reached,
+      };
+    });
+  };
+
+  const fetchProviders = async (instance_type: string): Promise<CreateKafkaInstanceWithSizesTypes.Providers> => {
+    try {
+      const apisService = getApi();
+      const res = await apisService.getCloudProviders();
+      const allProviders = res?.data?.items || [];
+      return await Promise.all(
+        allProviders
+          .filter((p) => p.enabled)
+          .map(async (provider): Promise<CreateKafkaInstanceWithSizesTypes.ProviderInfo> => {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const regions = await fetchRegions(provider.id!, instance_type);
+            return {
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              id: provider.id!,
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              displayName: provider.display_name!,
+              regions,
+            };
+          })
+      );
+    } catch (e) {
+      console.error('useAvailableProvidersAndDefault', 'fetchProvider', e);
+      return Promise.reject(e);
+    }
+  };
+
+  // TODO - this logic should not exist and be replaced with AMS quota.
```

Comment:
This logic used for trial kafka to restrict user to create single instance on create modal.
We can remove this logic and use AMS quota if  AMS quota set the limit for trial/developer. 
AFAIK, AMS quota doesn't set the limit for trail, they always have zero limit. 
Do know about AMS quota limit for trial/developer ?

Reply:
We cannot avoid having different logic for trial and standard instances sadly or use any AMS values for trial.

Reply:
> We can remove this logic and use AMS quota if AMS quota set the limit for trial/developer.

Would explore that idea